base-notebook/Dockerfile

```diff
@@ -49,7 +50,7 @@ USER jovyan
 # Setup jovyan home directory
 RUN mkdir /home/$NB_USER/work && \
     mkdir /home/$NB_USER/.jupyter && \
-    mkdir /home/$NB_USER/.local && \
+    mkdir -p -m 700 /home/$NB_USER/.local/share/jupyter && \
```

Comment:
If a user follows the directions in the readme to host mount a ssl certificate where we have preconfigured the notebook server to look for it, docker creates the intervening directories with root ownership and read only permissions. This effectively prevents the notebook server from writing other files into this directory like the notebook signing key and breaks everything.

We could change the notebook config to look for the pem file in another directory and break backward compatibility with older images. I opted for keeping the file where it has been and making the subdirectories ahead of time.